### ISSUE TITLE
LNSBT - [Track 2. Web3.0 Development Grant] 

### ISSUE THREAD

**Issue by lnsbt** [Waves-Association/grants-program#58]
- **Proposer:** Introduce yourself (name, surname) 
- Berkan Turan
- **Contacts:** Your email or telegram handle
- lnsbtcom@gmail.com
- **Github username:** Your Github login  
 https://github.com/lnsbt
 
## 👥  Team 

- Are you applying to implement the grant as an individual developer or do you represent a team?  
I'm applying as an individual.

### Team experience  (in case applying as a team or startup)

- Please describe your (team/startup) development experience and technological stack?  
    Smart Contract - Ride  
    Frontend - JavaScript  
    Backend - NodeJS  
    Database - MongoDB
- Do(es) you(r) (team/startup) have experience in using Waves technologies?  
    Yes, I have created 2 DeFi tools in Waves ecosystem,  
    https://swopscan.com (it is active over 1 year)  
    https://puzzlescan.com (stopped working on it at the beginning of December)
- Please attach your (team/startup) portfolio (projects you’ve been involved in)   
    https://swopscan.com  
    https://puzzlescan.com
- Team website  
    https://lnsbt.com

### Team Code Repos

- https://github.com/lnsbt

## 🔍 Overview

Please describe your solution idea by the following structure:

- Short description of the solution you suggest  
    At the time of writing, over 95% of NSBT is being staked. Stakers are regularly rewarded with the fee collected in the protocol.  
    This staked NSBT turns into an illiquid asset, which is around $35.000.000.  
    So to speak the issue here surely is not specific to NSBT itself. But LNSBT specifically targets the illiquid staked NSBT. <br />
- How does it solve the specified problem?  
    LNSBT is an experimental project with the target of possibly contributing to a better solution to illiquid asset issues, specifically illiquid staked NSBT. <br />
    There are various attempts to solve this problem on other networks, LNSBT design choice is different and is worth testing.  
  
    **How LNSBT works:**  
    I will try to briefly explain how LNSBT is planned to work for the first 2 rounds which are covered in this grant proposal.  
    
    &nbsp;&nbsp; -LNSBT is designed to run in rounds with a block range, total NSBT amount limit, and strategy.  
    &nbsp;&nbsp; -For each round:  
    &nbsp;&nbsp;&nbsp;&nbsp; -Users will be able to lock their NSBT until the target block of the round.  
    &nbsp;&nbsp;&nbsp;&nbsp; -Users will get a newly issued LNSBT_targetBlock(LNSBT specific round token) token. This LNSBT_targetBlock amount corresponds to NSBT they locked.  
    &nbsp;&nbsp;&nbsp;&nbsp; -LNSBT will immediately start staking the NSBT that users locked.  
    &nbsp;&nbsp;&nbsp;&nbsp; -LNSBT will also stake the Neutrino Protocol rewards, specifically USDN, and EURN if there are any.  
    &nbsp;&nbsp;&nbsp;&nbsp; -With the ending of the round, LNSBT_targetBlock owners can give their LNSBT_targetBlock and get the same amount of NSBT from the LNSBT smart contract.  
    &nbsp;&nbsp;&nbsp;&nbsp; -Only the users who locked their NSBT, have the right of claiming the rewards, specifically WAVES, USDN, and EURN if there are any.  
  
    One question may occur here; why LNSBT just distributes WAVES, USDN, and EURN but not the other possible rewards.  
    &nbsp;&nbsp; -This comes from a design choice of giving users a chance to involve and reward some of those users. <br />
    &nbsp;&nbsp; -Users will be able to call any function in the contract(if the conditions match) including the activation, staking rewards, ending, etc.  
    &nbsp;&nbsp; -This interaction chance will come with a possible reward, specifically "RUBN", "CNYN", "JPYN", "UAHN", "BRLN", "GBPN", "TRYN"(if there are any in the contract at that moment), will go to the user who calls the reward staking function.  
  
    NSBT locking motivations:  
    &nbsp;&nbsp; -Users who lock their NSBT will be guaranteed of having the possible staking rewards for the period of the round.  
    &nbsp;&nbsp; -Users will also have LNSBT_targetBlock to represent NSBT redeeming power at the end of the round.  
    &nbsp;&nbsp; -LNSBT_targetBlock, if it happens;  
    &nbsp;&nbsp;&nbsp;&nbsp; -Can be traded at secondary market freely,  
    &nbsp;&nbsp;&nbsp;&nbsp; -Can be used in liquidity farming,   
    &nbsp;&nbsp;&nbsp;&nbsp; -Can be used as collateral on lending protocols.

    LNSBT_targetBlock at secondary market  
    &nbsp;&nbsp; -Users who lock their NSBT will be guaranteed of having the staking reward for the period of the round.  
    &nbsp;&nbsp;&nbsp;&nbsp; -This can motivate some users to sell their LNSBT_targetBlock at a lower price than the NSBT price.  
    &nbsp;&nbsp;&nbsp;&nbsp; -This price difference is expected to be tied to the round duration, round limit, round strategy.  
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(This is one of the reasons to execute 2 rounds with different terms as this proposal suggests)  
    &nbsp;&nbsp; -This behavior can be an opportunity for long-term buyers to buy with a discount.  
    &nbsp;&nbsp; -These two motivations are expected to have a big effect on the price equilibrium in the secondary market.  
  
- How you will integrate this project into Waves / Neutrino.  
  LNSBT directly targets Neutrino Protocol.  
  
- Why your team is interested in creating this project.  
  I have a significant interest in technology and economy. Considering the enormous amount of illiquid assets, I want to run this experimental project to possibly contribute to a better solution to the problem. While I'm familiar with the Waves community and ecosystem, this is a perfect environment to execute this idea.  
  
- What is the target audience of the final solution?  
  NSBT stakers are the first group. If LNSBT successes in its arguments and continues its existence, it can be an important actor for illiquid assets in the Waves ecosystem and beyond.
  
- What is the “secret sauce” of it?  
  I'm familiar with the Waves community, ecosystem, projects, problems, expectations. With gathering those and giving the users great control over the functionalities of the dApp, I hope this will be a different experience on Waves.  
  
  I'm also experienced running data-related tools as I referred at the beginning, I will try to provide a good amount of data to the users, also to the other buidlers. As this data will help us to understand this experience, it can also be helpful for other buidlers. This is crucial because I know at the end, users will  either let LNSBT grow or die. Either way, the data will be there and be a potential guide for whoever wants to do better.  
  
  For the duration of this proposal suggests, there won't be any token other than NSBT corresponding token. This will bring real users who want to test and stay if they want to without any other motivation than the pure benefit of LNSBT design itself.  
  There won't be any worries about any token price, hype, or anything else. That will let me give my focus on LNSBT and successfully execute the first 2 rounds.
  
- What is the duration of the whole project?  
  LNSBT was successfully launched today with the necessary improvements after our mainnet test (it was not publicly announced, a group of individuals who were informed participated).  
    Until the release of the first round, the target is creating awareness for the project and more importantly the idea.  
      
    - The first round is planned to be released on January 5, 2022.  
    The round is planned as 1440*30 blocks with 2000 NSBT limit, without any individual limit.  
  
    - The second round is planned to run shortly after the end of the first round, mid of February 2022.  
    This round is planned as 1440*90 blocks with 4000 NSBT limit, without any individual limit.  
  
- Resources required to implement the grant task  
    For the duration of this proposal suggests, LNSBT won't have any fee, also there won't be any token other than NSBT corresponding token. This grant will be the only financial instrument that will help take the necessary steps.  
    As LNSBT was successfully launched today with months of work prior to this proposal, from now on LNSBT requires continuous development/maintenance and marketing/adoption.  
    With the help of this grant, I'd like to give my focus to LNSBT.  
    To be more precise:  
    I'd like to allocate my work, more or less:  
    &nbsp;&nbsp; -60% marketing, adoption.  
    &nbsp;&nbsp; -35% development, maintenance.  
    &nbsp;&nbsp; -5% of other matters related LNSBT.  
    If I need, I won't hesitate to bring extra force. In such a case, I will inform you in the related report. But at this point, I can run this with the help of the community and that will help me a lot to understand all the aspects. 
  
### Ecosystem Fit

Are there any other projects similar to yours? If so, how is your project different?  <br />
    There is no such project on Waves.  <br />
    Even they are totally different protocols than LNSBT, Swop and Vires have their own unique approaches in terms of sharing staking rewards with users. Swop users are vulnerable to impermanent loss and at the moment, NSBT is not on Vires.  <br />
    There are projects on other networks that work to solve the problem of illiquid assets. Also, some projects involved in Polkadot Parachain Auctions have different approaches in this matter. ETH 2.0 staking methods are quite teaching as well. <br /><br />
    LNSBT comes with its own approach with;  
      &nbsp;&nbsp; -Fixed time lock rounds,  
      &nbsp;&nbsp; -Incentivizing the locker,  
      &nbsp;&nbsp; -Possible discounted secondary market opportunity for those who want to buy,  
      &nbsp;&nbsp; -Possible higher rewards than individual staking by compounding the rewards.

## 🔩 Development Roadmap 

This section should break out the development roadmap into a number of milestones. It helps to describe the functionality we should expect, plus how we can check that such functionality exists in the product. Whenever milestones are delivered, we refer to the roadmap to ensure that everything has been delivered as expected.

**Phase #1**: December 2021 - February 2022 ($25,000)  
    As LNSBT was successfully launched today, the main focus will be marketing for making people familiar with the idea and the project.  
    Phase 1 is expected to happen with the following steps and KPI to deliver:  
    -Pitching the idea and the project on community channels with the opportunities it can bring, also the risks it carries on its own.  
    -The release of the first round which will last ~30 days with 2000 NSBT limit.  
    &nbsp;&nbsp; -The first round is planned to be released on January 5, 2022.  
    -Targetted KPI to deliver:  
    &nbsp;&nbsp;-TVL ~1500-2000 NSBT  

**Phase #2**:  February 2022 - May 2022 ($35,000)  
    With the guidance of Phase 1, the main focus will still be marketing for making people familiar with the idea and the project as it is still new and as the second round will be run with different conditions.  
    Phase 2 is expected to happen with the following steps and KPI to deliver:  
    -Continue pitching the idea and project as it will still be a new experience.  
    -The release of the second round which will last ~90 days with 4000 NSBT limit.  
    &nbsp;&nbsp; -The second round is planned to be released in mid of February 2022.  
    -Targetted KPI to deliver:  
    &nbsp;&nbsp;-TVL ~3000-4000 NSBT  

Within this application we are requesting **Phase #1** funding with a total cost of $25000 (or 1675 WAVES) and following KPIs to deliver:  
    As there won't be any individual limit but a total limit(2000 NSBT) in the contract for the first round, the target is reaching 75% of the limit with as many users as want to participate.  
    KPI:  
    &nbsp;&nbsp; -TVL ~1500-2000 NSBT
    
Within this application we are requesting **Phase #2** funding with a total cost of $35000 (or 2325 WAVES) and following KPIs to deliver:  
    As there won't be any individual limit but a total limit(4000 NSBT) in the contract for the second round, the target is reaching 75% of the limit with as many users as want to participate.  
    KPI:  
    &nbsp;&nbsp; -TVL ~3000-4000 NSBT

### Overview

- **Total Estimated Duration:** Duration of the whole project  
  ~5 months
- **Total Costs:** Amount of Payment in Waves for the whole project. The total amount of funding needs to not more than 100k Waves.  
  4000WAVES

## Additional Information ➕

Any additional information that you think is relevant to this application that hasn't already been included.
  

<img src="https://i.imgur.com/02iyxw5.png" />
1) USER A locks his NSBT in LNSBT smart contract.  <br />
   Contract reissues the same amount of LNSBT_.(. represents the block height for this round) and sends it to USER A  <br />
2) LNSBT smart contract stakes the locked NSBT in NEUTRINO smart contract.  <br />
3) NSBT staking rewards are regularly sent to LNSBT smart contract.  <br />
4) LNSBT smart contract:  <br />
     &nbsp;&nbsp;Keeps WAVES rewards in the account,  <br />
     &nbsp;&nbsp;Stakes USDN, and EURN in NEUTRINO smart contract(s)  <br />
     &nbsp;&nbsp;"RUBN", "CNYN", "JPYN", "UAHN", "BRLN", "GBPN", "TRYN" rewards are sent to this function caller  <br />
5) USDN and EURN staking rewards are sent to LNSBT smart contract.  <br />
   USDN and EURN rewards are staked in NEUTRINO smart contract(s) similar to step 4, together with NSBT rewards (explained in step 4).  <br />
6) USER A freely exchanges LNSBT_. token with USER B  <br />
7) At the end of the round, staked tokens (NSBT, USND, EURN) are unstaked.  <br />
8) USER B unlocks NSBT from LNSBT smart contract with LNSBT_. he owns.  <br />
9) USER A unlocks NSBT from LNSBT smart contract with LNSBT_. he owns.  <br />
10) USER A gets his locking rewards(WAVES, USDN, EURN) from the LNSBT smart contract after the cooling period(2880 blocks from the ending block).  <br />

**Comment by BlackTurtle123** [comment]
Hello,
We discussed your application and came with some feedback:
=> you need a multisig wallet with community members to make it more secure and less trust needed for everyone
=> please use smaller milestones (easier to evaluate)
=> we will not award phase 1 but use it as a phase to evaluate the project
=>kpi's are to low
=> we will also want to wait for neutrino updates till they are released and look how your project will interact with those

**Comment by lnsbt** [comment]
Hello,

Thank you all members who evaluated the proposal and provided the feedback.

=> you need a multisig wallet with community members to make it more secure and less trust needed for everyone
Thank you for the feedback.
In Waves Developer Telegram group(waves_ride_dapps_dev), I publicly offered community multisig wallet 2 times (messages are there). Nobody did not respond. I respected, moved on, didn't look back, won't look back, time is crucial. 
I used my time efficiently and developed another strategy for possible future LNSBT rounds, it is being tested on the mainnet now. 
<img src="https://i.imgur.com/SWgcnMh.png" />

=> please use smaller milestones (easier to evaluate)
Thank you for the feedback.
Milestones were carefully thought and they are specific to the nature of the idea, especially the idea of certain time-lock of the rounds.

=> we will not award phase 1 but use it as a phase to evaluate the project
Thank you for the feedback.

 =>kpi's are to low
Thank you for the feedback.
KPI is around ~0.1%-~0.2% of circulating supply. These numbers are for an experimental idea which is expected to move fast and test different conditions. 
As an example, even it is not in this proposal, I developed a different strategy(I mentioned above) to make the idea easier to be understood and possibly being more beneficial for users.
First round has to stay with the same limits, 2000NSBT. It is a matter of trust as I already announced the conditions. Cannot be changed.
Second round will be shaped with the users' needs.

 => we will also want to wait for neutrino updates till they are released and look how your project will interact with those
Thank you for the feedback.
It is a way of thinking and approaching to the issues. I saw an issue of the moment, tried to offer a solution. 
Future's solutions won't take away today's problems.

I kindly invite the members to reevaluate the proposal.
Thank you all for your valuable time.

**Comment by BlackTurtle123** [comment]
Hello, we had a new meeting:

=> Update in 3 weeks, we will wait for this
=> We are dissatisfied with the multisig answer. There is a multisig solution being built

**Comment by lnsbt** [comment]
Hello,

LNSBT first round has been successfully executed. 
The first round data can found at [https://medium.com/@lnsbt/lnsbt-2973780-summary-5abdbd1d885b](https://medium.com/@lnsbt/lnsbt-2973780-summary-5abdbd1d885b).

As the result of the community voting in our Telegram group, it was decided that the possible second round will be discussed with the community after the long-awaited Neutrino Protocol upgrade. 

Thank you all for your valuable time.

**Comment by BlackTurtle123** [comment]
> Hello,
> 
> LNSBT first round has been successfully executed. The first round data can found at https://medium.com/@lnsbt/lnsbt-2973780-summary-5abdbd1d885b.
> 
> As the result of the community voting in our Telegram group, it was decided that the possible second round will be discussed with the community after the long-awaited Neutrino Protocol upgrade.
> 
> Thank you all for your valuable time.

Hello!
Please give us your updated plans for lnsbt, as we understand the staking fee is pretty high as is, so short round don't seem to work anymore.

Hope to hear back from you.

Regards

**Comment by lnsbt** [comment]
Hello,

After the new Neutrino Protocol upgrade details, I and other LNSBT contributors came up with a solution, thanks to all LNSBT contributors, that will surpass the unstaking fine with an optimal market efficiency.
I offered the proposal in various platforms and did the main development of the core of it.

When I first started, I had hopes of building with all the parties of Waves, but during last couple of months I felt more like building against them. 
Because of this, in the proposal, I said I would develop the new version and give it away to another team in the Waves ecosystem that is capable of further development, running and trusted. 

During that time no team approached me.
As how it was decided to not run the second round, in the Telegram group, contributors voted "No" to whether or not they would be in if any other team run the idea.

I always respected their decisions, and I do again. After the end of voting and result of not being approached by any other team or individual, it was decided to end LNSBT experience for now.

Thank you all for your valuable time.
